apps/smart-contracts/core/test/WithdrawHook.test.ts

```diff
@@ -195,4 +231,279 @@ describe('=> WithdrawHook', () => {
       await expect(tx).to.emit(withdrawHook, 'DepositRecordChange').withArgs(depositRecord.address)
     })
   })
+
+  describe('# setWithdrawalsAllowed', () => {
+    it('reverts if not role holder', async () => {
+      expect(
+        await withdrawHook.hasRole(await withdrawHook.SET_WITHDRAWALS_ALLOWED_ROLE(), user.address)
+      ).to.eq(false)
+
+      await expect(withdrawHook.connect(user).setWithdrawalsAllowed(true)).revertedWith(
+        `AccessControl: account ${user.address.toLowerCase()} is missing role ${await withdrawHook.SET_WITHDRAWALS_ALLOWED_ROLE()}`
+      )
+    })
+
+    it('sets to false', async () => {
+      await withdrawHook.connect(deployer).setWithdrawalsAllowed(true)
+      expect(await withdrawHook.withdrawalsAllowed()).to.not.eq(false)
+
+      await withdrawHook.connect(deployer).setWithdrawalsAllowed(false)
+
+      expect(await withdrawHook.withdrawalsAllowed()).to.eq(false)
+    })
+
+    it('sets to true', async () => {
+      expect(await withdrawHook.withdrawalsAllowed()).to.not.eq(true)
+
+      await withdrawHook.connect(deployer).setWithdrawalsAllowed(true)
+
+      expect(await withdrawHook.withdrawalsAllowed()).to.eq(true)
+    })
+
+    it('is idempotent', async () => {
+      expect(await withdrawHook.withdrawalsAllowed()).to.not.eq(true)
+
+      await withdrawHook.connect(deployer).setWithdrawalsAllowed(true)
+
+      expect(await withdrawHook.withdrawalsAllowed()).to.eq(true)
+
+      await withdrawHook.connect(deployer).setWithdrawalsAllowed(true)
+
+      expect(await withdrawHook.withdrawalsAllowed()).to.eq(true)
+    })
+
+    it('should emit a WithdrawalsAllowedChange event', async () => {
```

Comment:
```suggestion
    it('emits WithdrawalsAllowedChange', async () => {
```
same for other event-emitting tests